controllers/integration/integration_controller.go

```diff
@@ -199,6 +201,26 @@ func setupApplicationSnapshotCache(mgr ctrl.Manager) error {
 		"spec.application", applicationSnapshotIndexFunc)
 }
 
+// setupEnvironmentCache adds a new index field to be able to search Environments by Application.
+func setupEnvironmentCache(mgr ctrl.Manager) error {
+	environmentIndexFunc := func(obj client.Object) []string {
+		return []string{obj.(*appstudioshared.ApplicationSnapshotEnvironmentBinding).Spec.Application}
+	}
+
+	return mgr.GetCache().IndexField(context.Background(), &appstudioshared.Environment{},
```

Comment:
Ran into an error on this line, fixed it by changing to:
```
return mgr.GetCache().IndexField(context.Background(), &appstudioshared.ApplicationSnapshotEnvironmentBinding{},
```

Reply:
Updated

Reply:
Thanks!